tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -81,15 +90,24 @@ get_interval_from_config(const Dimension *dim, const Jsonb *config, const char *
 }
 
 int64
-policy_refresh_cagg_get_refresh_start(const Dimension *dim, const Jsonb *config, bool *start_isnull)
+policy_refresh_cagg_get_refresh_start(const Dimension *dim, const Jsonb *config)
 {
-	return get_interval_from_config(dim, config, CONFIG_KEY_START_INTERVAL, start_isnull);
+	bool start_isnull;
+	int64 res = get_interval_from_config(dim, config, CONFIG_KEY_START_INTERVAL, &start_isnull);
+	/* interpret NULL as min value for that type */
+	if (start_isnull)
+		return ts_time_get_min(ts_dimension_get_partition_type(dim));
+	return res;
 }
 
 int64
-policy_refresh_cagg_get_refresh_end(const Dimension *dim, const Jsonb *config, bool *end_isnull)
+policy_refresh_cagg_get_refresh_end(const Dimension *dim, const Jsonb *config)
 {
-	return get_interval_from_config(dim, config, CONFIG_KEY_END_INTERVAL, end_isnull);
+	bool end_isnull;
+	int64 res = get_interval_from_config(dim, config, CONFIG_KEY_END_INTERVAL, &end_isnull);
+	if (end_isnull)
+		return ts_time_get_max(ts_dimension_get_partition_type(dim));
```

Comment:
I think we need to revisit the "defaults" when we input NULL values. It makes more sense to me that NULL end_interval means "current time" (i.e., no offset) instead of infinite refresh.

I we keep this interpretation o NULL, we should return "end" if possible, otherwise we won't refresh the whole time range for timestamp types (given that the refresh window is exclusive). Also, would be good to get this line covered by a test.

```suggestion
		return ts_time_get_end_or_max(ts_dimension_get_partition_type(dim));
```

Reply:
I'll change this to use ts_time_get_end_or_max. We use start_interval=NULL to represent -infinity. So makes sense to use that for end_interval as well to represent +infinity, instead of arbitrarily clamping it to current time. (This means the window could include data from the "future").  Wouldn't this translate to the max time value in the data anyway since there is nothing else to process?